modules/core/src/main/java/org/apache/ignite/internal/processors/cache/consistentcut/ConsistentCutFuture.java

```diff
@@ -0,0 +1,31 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.ignite.internal.processors.cache.consistentcut;
+
+import java.util.UUID;
+import org.apache.ignite.internal.IgniteInternalFuture;
+import org.apache.ignite.internal.pagemem.wal.record.ConsistentCutFinishRecord;
+import org.apache.ignite.internal.processors.cache.persistence.wal.WALPointer;
+
+/**
+ * Future that completes after Consistent Cut finished and return pointer to {@link ConsistentCutFinishRecord}.
+ */
+public interface ConsistentCutFuture extends IgniteInternalFuture<WALPointer> {
+    /** Consistent Cut ID. */
+    public UUID id();
```

Comment:
Can we store id outside of `ConsistentCutFuture` and get rid of ConsistentCutFuture interface and ClientConsistentCutFuture?